src/visualization/convert.jl

```diff
@@ -73,6 +73,44 @@ function get_data_2d(center_level_0, length_level_0, leaf_cells, coordinates, le
 end
 
 
+# Extract data from a 1D DG solution and prepare it for visualization as a line plot.
+# This returns a tuple with
+# - x coordinates
+# - nodal 1D data
+#
+# Note: This is a low-level function that is not considered as part of Trixi's interface and may
+#       thus be changed in future releases.
+function get_data_1d(original_nodes, unstructured_data, nvisnodes)
+  # Get the dimensions of u; where n_vars is the amount of variables, a the amount of entries in each element and n_nodes the amount of elements.
+  n_nodes, n_elements, n_vars = size(unstructured_data)
+
+  # Set the amount of nodes visualized according to nvisnodes.
+  if nvisnodes === nothing
+    max_nvisnodes = 2 * n_nodes
+  elseif nvisnodes == 0
+    max_nvisnodes = n_nodes
+  else
+    max_nvisnodes = nvisnodes
+  end
+
+  interpolated_nodes = Array{Float64,2}(undef, max_nvisnodes, n_elements)
+  interpolated_data = Array{Float64,2}(undef, max_nvisnodes*n_elements, n_vars)
+
+  # Iterate over all elements.
+  for i=1:n_elements
+    # Interpolate on an equidistant grid.
+    interpolated_nodes[:,i] = vcat(range(original_nodes[1,1,i], original_nodes[1,end,i], length = max_nvisnodes))
+
+    # Interpolate the data for each variable.
+    for j=1:n_vars
+      interpolated_data[(i-1)*max_nvisnodes+1:i*max_nvisnodes, j] = interpolate1d(original_nodes[1,:,i], unstructured_data[:,i,j], interpolated_nodes[:,i])
```

Comment:
Why don't you make `interpolated_data` a 3-dimensional array first to make this line easier to read, and then return it below as `reshape(interpolated_data, max_nvisnodes*n_elements, n_vars)`?

Reply:
This is a great idea, but with the fix on the "nvisnodes = 1"-issue, there are now max_nvisnodes*n_elements+1 nodes (just like in 2D). This sadly cannot be put into a 3D array anymore.

I tried making it more readable in 916e622 by introducing list of indices instead of just putting them in like this.

Reply:
Hm, you could try to avoid this by just requiring `nvisnodes = 2` as the minimum. It is a small restriction, but OTOH I don't think that it is that problematic for 1D, and it would make the code much more legible.

Reply:
I updated this in a2e1092.
Now the matrix is 3-dimensional at first and then reshaped.
I also then appended the last entry to the reshaped matrix for each variable, which means that nvisnodes=1 still works.

Reply:
If

> nvisnodes=1 still works

why do we have
```julia
@assert nvisnodes >= 2 "nvisnodes must be zero or >= 2"
```

Reply:
Sorry, this is an old, unresolved thread. Since we interpolate each element using the end points, the minimum is indeed 2.